mmengine/utils/progressbar_rich.py

```diff
@@ -0,0 +1,214 @@
+            self.completed += advance
+            elapsed = self.timer.since_start()
+            if elapsed > 0:
+                fps = self.completed / elapsed
+            else:
+                fps = float('inf')
+            self.write(
+                f'completed: {self.completed}, '
+                f'elapsed: {int(elapsed + 0.5)}s, '
+                f'{fps:.1f} tasks/s', self.colors[task_id])
+        else:
+            assert task_id >= -1 and task_id < len(self.tasks), (
+                'The task_id must be within a valid range')
+
+            # Activate self.bar before all tasks have been updated.
+            if all(task.completed == 0 for task in self.bar.tasks):
+                self.bar.start()
+
+            if task_id == -1:
+                task_id = self.bar.tasks[-1].id
+
+            completed = self.bar.tasks[task_id].completed + advance
+            total = self.bar.tasks[task_id].total
+            self.bar.update(
+                task_id,
+                advance=advance,
+                description=f'[{self.colors[task_id]}]'
+                f'{self.descriptions[task_id]}'
+                f'_{completed}/{total}')
+
+            # After all tasks are completed, deactivate self.bar.
+            if all(task.finished for task in self.bar.tasks):
+                self.bar.stop()
+
+
+def worker(params, function):
+    result = function(*params)
+    return result
+
+
+def track_progress_rich(func: Callable,
+                        tasks: Tuple[list],
+                        nproc: int = 1,
+                        description: str = 'process...',
+                        color: str = 'blue',
+                        chunksize: int = 1,
+                        skip_first: bool = False,
+                        keep_order: bool = True) -> list:
+    """Track the progress of parallel task execution with a progress bar.
+
+    The built-in :mod:`multiprocessing` module is used for process pools and
+    tasks are done with :func:`Pool.map` or :func:`Pool.imap_unordered`.
+
+    Args:
+        func (callable): The function to be applied to each task.
+        tasks (tuple[list]): A tuple of tasks.
+        nproc (int): Process (worker) number, if nuproc is 1, use single
+            process. Default is 1.
+        description (str): The description of progress bar.
+        color (str): The color of progress bar.
+        chunksize (int): Refer to :class:`multiprocessing.Pool` for details.
+        skip_first (bool): Whether to skip the first sample for each worker
+            when estimating fps, since the initialization step may takes
+            longer.
+        keep_order (bool): If True, :func:`Pool.imap` is used, otherwise
+            :func:`Pool.imap_unordered` is used.
+
+    Returns:
+        list: The task results.
+    """
+    assert is_seq_of(tasks,
+                     Iterable), 'The content of tasks must be iterable object'
+    assert nproc > 0, 'nproc must be a positive number'
```

Comment:
According to [google python style guide](https://google.github.io/styleguide/pyguide.html#244-decision), we should not use assert to raise an error, but raise the specific error.